src/actions/indexing/indexable-post-indexation-action.php

```diff
@@ -124,23 +126,67 @@ public function get_limit() {
 	 * @return string The query.
 	 */
 	protected function get_query( $count, $limit = 1 ) {
+		if ( $count ) {
+			return $this->get_count_query( $limit );
+		}
+		return $this->get_select_query( $limit );
+	}
+
+	/**
+	 * Builds a query for counting the number of unindexed posts.
+	 *
+	 * @param bool $limit The maximum amount of unindexed posts that should be counted.
+	 *
+	 * @return string The prepared query string.
+	 */
+	protected function get_count_query( $limit = false ) {
 		$indexable_table = Model::get_table_name( 'Indexable' );
 		$post_types      = $this->get_post_types();
 		$replacements    = $post_types;
 
-		$select = 'P.ID';
-		if ( $count ) {
-			$select = 'COUNT(P.ID)';
+		$limit = '';
+		if ( $limit ) {
+			$limit_query    = 'LIMIT %d';
+			$replacements[] = $limit;
 		}
-		$limit_query = '';
-		if ( ! $count ) {
+
+		// Warning: If this query is changed, makes sure to update the query in get_select_query as well.
+		return $this->wpdb->prepare( "
+			SELECT COUNT(P.ID)
```

Comment:
@Xyfi For the "`select ID's (LIMIT shutdown_limit) order by ID ` instead of counting the entire indexables in the "count" queries" item, I think we should:
```suggestion
			SELECT P.ID
```
here, so that the limit can actually have an effect. (see: https://stackoverflow.com/a/17020968)

And if we do that, we should then count the result array with PHP.

We only care if the count is over the limit, so if we limit a SELECT with limit+1, and then check if that result contains over 25 rows, we're good to continue.